src/main/kotlin/org/wysko/midis2jam2/world/camera/AutoCamController.kt

```diff
@@ -286,431 +311,456 @@ enum class AutoCamPosition(
         AutoCamPositionType.INSTRUMENT
     ),
 
+    DRUM_SET(
+        Vector3f(-0.2f, 61.6f, 38.6f),
+        Quaternion(-5.8945218E-9f, 0.9908659f, -0.13485093f, -4.3312124E-8f),
+        { _, _, context -> context.drumSetVisibilityManager.isVisible },
+        { _, _, context -> context.drumSetVisibilityManager.isVisible },
+        AutoCamPositionType.INSTRUMENT
+    ),
+
+    KEYBOARDS_2(
+        Vector3f(0f, 71.8f, 44.5f),
+        Quaternion().fromAngles(-2.867576f, 0.7836529f, -3.1415927f),
+        { time, instruments, context -> context.configs.getType(org.wysko.midis2jam2.starter.configuration.SettingsConfiguration::class).isClassicCamera && visibleNowAndLater(instruments, Keyboard::class.java, time, WAIT_TIME * 1.5) },
+        { _, instruments, _ -> instruments.filterIsInstance<Keyboard>().any { it.isVisible } },
+        AutoCamPositionType.INSTRUMENT
+    ),
+
+    DRUM_SET_2(
+        Vector3f(-19.6f, 78.7f, 3.8f),
+        Quaternion().fromAngles(-2.6581365f, -0.2827433f, 3.1415927f),
+        { _, _, context -> context.configs.getType(org.wysko.midis2jam2.starter.configuration.SettingsConfiguration::class).isClassicCamera && context.drumSetVisibilityManager.isVisible },
+        { _, _, context -> context.drumSetVisibilityManager.isVisible },
+        AutoCamPositionType.INSTRUMENT
+    ),
+
+    BASS_GUITAR_2(
+        Vector3f(35f, 25.4f, -19f),
+        Quaternion().fromAngles(2.268928f, -1.0646509f, 3.0979594f),
+        { time, instruments, context -> context.configs.getType(org.wysko.midis2jam2.starter.configuration.SettingsConfiguration::class).isClassicCamera && visibleNowAndLater(instruments, BassGuitar::class.java, time, WAIT_TIME * 1.5) },
+        { _, instruments, _ -> instruments.filterIsInstance<BassGuitar>().any { it.isVisible } },
+        AutoCamPositionType.INSTRUMENT
+    ),
+
     SOPRANO_SAX(
         Vector3f(18.91f, 40.76f, -11.10f),
         Quaternion(-0.04f, 0.96f, -0.19f, -0.19f),
-        { time, instruments, _ -> visibleNowAndLater(instruments, SopranoSax::class.java, time, WAIT_TIME * 1.5) },
+        { time, instruments, context -> !context.configs.getType(SettingsConfiguration::class).isClassicCamera && visibleNowAndLater(instruments, SopranoSax::class.java, time, WAIT_TIME * 1.5) },
```

Comment:
Instead of plaguing each of the instrument entries, add a new property to `AutoCamPosition` called something like `isClassicCam`, default it to `false`, set it to `true` for the right positions, then filter the list of appropriate positions based on that property when selecting a random camera.

Furthermore, logic in the `randomCamera` function is unchanged, so it will still try to pick an instrument camera  whereas the classic camera positions are only stage cameras. So I think it's failing and giving up, which is why I think the classic cam is not moving often (just waiting to try again).